doc/cookbook/guard_authentication.rst

```diff
@@ -0,0 +1,150 @@
+How to Create a Custom Authentication System with Guard
+=======================================================
+
+Whether you need to build a traditional login form, an API token authentication system or you need to integrate with some proprietary single-sign-on system, the Guard component can make it easy... and fun!
+
+In this example, you'll build an API token authentication system and learn how to work with Guard.
+
+Step 1) Create the Authenticator Class
+--------------------------------------
+
+Suppose you have an API where your clients will send an X-AUTH-TOKEN header on each request. This token is composed of the username followed by a password, separated by a colon. Your job is to read this, find the associated user (if any) and check the password.
```

Comment:
I think the standard is to add line breaks after about the 72nd character.

Also, you should give an example if the header:

```rst
the username followed by a password, separated by a colon (e.g. ``X-AUTH-TOKEN: coolguy:awesomepassword``)
```